packages/backend-common/src/config.ts

```diff
@@ -151,11 +180,20 @@ export async function loadBackendConfig(options: {
   const args = parseArgs(options.argv);
   const configPaths: string[] = [args.config ?? []].flat();
 
-  const config = new ObservableConfigProxy(options.logger);
-
   /* eslint-disable-next-line no-restricted-syntax */
   const paths = findPaths(__dirname);
 
+  // TODO(hhogg): This is fetching _all_ of the packages of the monorepo
+  // in order to find the secrets for redactions, however we only care about
+  // the backend ones, we need to find a way to exclude the frontend packages.
+  const { Project } = require('@lerna/project');
```

Comment:
This change broke our build 😢

We are managing multiple backstage apps/backends in a single mono repository, where not all apps use the same set of plugins. With this code, the backend crashes on startup due to a config constraint from an unrelated plugin.

**Example:**
1. `packages/backend-1` -> uses `plugins/plugin-a`
2. `packages/backend-2` -> uses `plugins/plugin-a` & `plugins/plugin-b`
3. `plugins/plugin-b` has the following `config.d.ts`:
```ts
export interface Config {
  someRequiredConfig: {
  /**
   * @visibility secret
   */
    secretValue: string;
  };
};
```

When we start the `backend-1` with `yarn workspace backend-1 start-backend`, it fails with the following error:
```
Backend failed to start up Error: Config validation failed, Config should have required property 'someRequiredConfig' { missingProperty=someRequiredConfig }
```

But the point is, that `packages/backend-1` has **NO** dependency to `plugins/plugin-b` so it shouldn't be forced to fulfil any of its config schemas. This is of course primarily a problem in the local build, since the dependency is not copied to the docker container, and the code won't find the config schema. But that's still a bit inconvenient to find errors locally. We now fell back to to using required configs for required fields.

The config validation uses the `--package` flag of the `backstage-cli config:*` commands to cope with this scenario, so it would be cool if this is also supported in the backend code. /cc @Rugvip